File: SetIntersection/setIntersect.rb
# set intersection soln in rb for codeeval.com by steven a dunn

require 'set'

File.open(ARGV[0], 'r').each_line do |line|
    params = line.split(";")

    set1 = params[0].chomp.split(',').to_set
    set2 = params[1].chomp.split(',').to_set

    intersection = set1 & set2

    puts intersection.to_a.join(",")
end

File: SetIntersection/py3/setIntersect.py
# Set Intersection solution in Python 3 for CodeEval.com by Steven A. Dunn

import sys

for line in open(sys.argv[1], 'r'):
  params = line.rstrip('\n').split(';')

  set1 = set(params[0].split(','))
  set2 = set(params[1].split(','))

  intersection = set1.intersection(set2)

  intersection = list(map(int, list(intersection)))
  intersection.sort()

  if len(intersection) == 0:
    print()
  elif len(intersection) == 1:
    print(intersection[0])
  else:
    for i in range(0, len(intersection) - 1):
      sys.stdout.write("%d," % intersection[i])

    print(intersection[-1])
